### PR TITLE
Add Windows Support

### PR DESCRIPTION
**Pull request by trhinehart-godaddy**
The proof is in the pudding!

```sh
> @godaddy/tartufo-node@0.3.2 precommit:tartufo C:\github\tartufo-node
> node ./bin/tartufo-shim.js pre-commit

Time: 2023-01-25T17:11:41.566935
All clear. No secrets detected.
[windows-support ca17854] Add Windows Support
 13 files changed, 52 insertions(+), 50 deletions(-)
 create mode 100644 lib/is-windows.js
Done
```

### CODE REVIEW

File: lib/python-local.js [hunk 1]
```diff
@@ -10,18 +9,16 @@ const { VENV_BIN_PATH } = require("./venv");
...
+    await access(VENV_PYTHON3_PATH, fs.constants.X_OK);
+    return VENV_PYTHON3_PATH;
   } catch (e) {
-    // noop
+    // ignore
   }
 
-  const pythonPath = path.join(VENV_BIN_PATH, "python");
   try {
-    await access(python3Path, fs.constants.X_OK);
```
Comment (trhinehart-godaddy) [inline lib/python-local.js:23]: Looks like this was a bug, was checking `python3` again so would always fail

File: tartufo.toml [hunk 2]
```diff
@@ -4,18 +4,21 @@ cleanup = true
 regex = true
 entropy = true
 
-exclude-path-patterns = [
-  'tartufo\.toml$',
-  'package-lock\.json$',
-  'CHANGELOG\.md$'
-]
+  [[tool.tartufo.exclude-path-patterns]]
+  path-pattern = "tartufo\\.toml$"
```
Comment (trhinehart-godaddy) [inline tartufo.toml:8]: this change was required, was getting an error `tartufo.types.ConfigException: String pattern is illegal in exclude-path-patterns`

Comment (dcousineau) [reply tartufo.toml:8]: I have #17 and #18 to hopefully address any needs here.

File: lib/venv.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
 const path = require("path");
+const isWin = require("./is-windows");
 
 const VENV_PATH = path.resolve(path.join(__dirname, "../python_venv"));
-const VENV_BIN_PATH = path.join(VENV_PATH, "bin");
+const VENV_BIN_PATH = path.join(VENV_PATH, isWin ? "Scripts" : "bin");
```
Comment (dcousineau) [inline lib/venv.js:5]: 🤔 very weird that this is different between WIN and POSIX. Thanks for finding!

**Review by dcousineau — APPROVED** [review]
Tentatively approving. I'm going to see about adding Windows to the matrix builds in this PR real quick before merging

**Comment by dcousineau** [comment]
Actually this is from a fork lol so I'm going to merge it and add in the windows matrix after.

**Pull request merged**